pkg/agent/git/clone.go

```diff
@@ -56,12 +108,35 @@ func CloneRepository(config *config.Config, project *serverapiclient.Project, au
 	return err
 }
 
+func fetchOrigin(repo *git.Repository, refSpecStr string, remote *git.Remote) error {
+	var refSpecs []gitconfig.RefSpec
+	if refSpecStr != "" {
+		refSpecs = []gitconfig.RefSpec{gitconfig.RefSpec(refSpecStr)}
+	}
+
+	if err := remote.Fetch(&git.FetchOptions{
+		RefSpecs: refSpecs,
+		Progress: os.Stdout,
+	}); err != nil {
+		if err == git.NoErrAlreadyUpToDate {
+			fmt.Print("refs already up to date")
+		} else {
+			return fmt.Errorf("fetch origin failed: %v", err)
+		}
+	}
+	return nil
+}
+
+func shouldClonePR(project *serverapiclient.Project) bool {
+	return project.Repository.PrNumber != nil
+}
+
 func shouldCloneBranch(project *serverapiclient.Project) bool {
-	if project.Repository.Branch == nil {
+	if project.Repository.Branch == nil || *project.Repository.Branch == "" {
```

Comment:
Checking for nil and "" handles all empty cases for string fields in [Repository](https://github.com/daytonaio/daytona/blob/main/pkg/types/workspace.go#L11) struct. 

This also fixes #326

Reply:
Let me investigate it, till then.